hazelcast/src/main/java/com/hazelcast/internal/iteration/IndexIterationPointer.java

```diff
@@ -139,13 +173,18 @@ public boolean isDescending() {
         return (flags & FLAG_DESCENDING) != 0;
     }
 
+    public Data getLastEntryKeyData() {
+        return lastEntryKeyData;
+    }
+
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
         out.writeByte(flags);
         out.writeObject(from);
         if ((flags & FLAG_POINT_LOOKUP) == 0) {
             out.writeObject(to);
         }
+        out.writeByteArray(lastEntryKeyData.toByteArray());
```

Comment:
```suggestion
        IOUtil.writeData(out, lastEntryKeyData);
```
I'm also surprised that this doesn't fail because `lastEntryKeyData` can be null.